R/summary_funs.R

```diff
@@ -170,230 +178,214 @@ weighted_summary_means <- function(y_wobs_test, family, wdraws, mu, dis, cl_ref,
-## actual (non-relative) value is computed. NOTE: Element `wcv[i]` (with i = 1,
-## ..., N and N denoting the number of observations) contains the weight of the
-## CV fold that observation i is in. In case of varsel() output, this is `NULL`.
-## Currently, these `wcv` are nonconstant (and not `NULL`) only in case of
-## subsampled PSIS-LOO CV. The actual observation weights (specified by the
+## The actual observation weights (specified by the
 ## user) are contained in `y_wobs_test$wobs`. These are already taken into
 ## account by `<refmodel_object>$family$ll_fun()` (or
 ## `<refmodel_object>$family$latent_ll_oscale()`) and are thus already taken
 ## into account in `lppd`. However, `mu` does not take them into account, so
 ## some further adjustments are necessary below.
-get_stat <- function(mu, lppd, y_wobs_test, stat, mu.bs = NULL, lppd.bs = NULL,
-                     wcv = NULL, alpha = 0.1, ...) {
-  n_notna.bs <- NULL
-  if (stat %in% c("elpd", "mlpd", "gmpd")) {
-    if (!is.null(lppd.bs)) {
-      # Compute the performance statistics using only those observations for
-      # which both `lppd` and `lppd.bs` are not `NA`:
-      lppd[is.na(lppd.bs)] <- NA
-      lppd.bs[is.na(lppd)] <- NA
-      n_notna.bs <- sum(!is.na(lppd.bs))
-    }
-    n_notna <- sum(!is.na(lppd))
-    n <- length(lppd)
-  } else {
-    hasNA_y <- is.na(y_wobs_test$y_prop %||% y_wobs_test$y)
-    if (!is.null(mu.bs)) {
-      # Compute the performance statistics using only those observations for
-      # which both `mu` and `mu.bs` are not `NA`:
-      mu[is.na(mu.bs)] <- NA
-      mu.bs[is.na(mu)] <- NA
-      n_notna.bs <- sum(!is.na(mu.bs) & !hasNA_y)
-    }
-    n_notna <- sum(!is.na(mu) & !hasNA_y)
-    n <- length(mu)
-  }
-  if (!is.null(n_notna.bs) && getOption("projpred.additional_checks", FALSE)) {
-    stopifnot(n_notna == n_notna.bs)
-  }
-  if (n_notna == 0) {
+get_stat <- function(summaries, summaries_baseline = NULL,
+                     summaries_fast = NULL,
+                     y_wobs_test, stat, alpha = 0.1, ...) {
+  mu <- summaries$mu
+  lppd <- summaries$lppd
+  loo_inds <- which(!is.na(lppd))
+  n <- length(lppd)
+  n_loo <- length(loo_inds)
+  if (n_loo == 0) {
     return(list(value = NA, se = NA, lq = NA, uq = NA))
   }
-
-  if (is.null(wcv)) {
-    ## set default CV fold weights if not given
-    wcv <- rep(1, n)
-  }
-  ## ensure the CV fold weights sum to n_notna
-  wcv <- n_notna * wcv / sum(wcv)
-
   alpha_half <- alpha / 2
   one_minus_alpha_half <- 1 - alpha_half
 
   if (stat %in% c("elpd", "mlpd", "gmpd")) {
-    if (!is.null(lppd.bs)) {
-      value <- sum((lppd - lppd.bs) * wcv, na.rm = TRUE)
-      value.se <- weighted.sd(lppd - lppd.bs, wcv, na.rm = TRUE) *
-        sqrt(n_notna)
+    if (is.null(summaries_baseline)) {
+      lppd_baseline = 0
+    } else {
+      lppd_baseline = summaries_baseline$lppd
+    }
+    if (!is.null(summaries_fast) && n_loo<n) {
```

Comment:
If `n_loo<n`, then `summaries_fast` should always be non-`NULL`, right? If that's correct, then we could simplify this `if` condition here.

Reply:
Checking this again, I'm pretty convinced we can perform this simplification. I'll do so in an upcoming commit.